app/scripts/lib/notification-manager.js

```diff
@@ -0,0 +1,74 @@
+const extension = require('extensionizer')
+const height = 520
+const width = 360
+
+
+class NotificationManager {
+
+  //
+  // Public
+  //
+
+  show () {
+    this.getPopup((err, popup) => {
+      if (err) throw err
+
+      if (popup) {
+        // bring focus to existing popup
+        extension.windows.update(popup.id, { focused: true })
+      } else {
+        // create new popup
+        extension.windows.create({
+          url: 'notification.html',
+          type: 'popup',
+          width,
+          height,
+        })
+        .catch((reason) => {
+          log.error('failed to create poupup', reason)
+        })
+      }
+    })
+  }
+
+  getPopup (cb) {
```

Comment:
This seems to be private, so would be nice to underscore-prefix, since we're defining a new protocol here, it's valuable to clearly demonstrate what's required to create a new platform's notification manager.